homeassistant/components/xiaomi.py

```diff
@@ -57,9 +57,21 @@ def _validate_conf(config):
 
 def setup(hass, config):
     """Set up the Xiaomi component."""
-    gateways = config[DOMAIN][CONF_GATEWAYS]
-    interface = config[DOMAIN][CONF_INTERFACE]
-    discovery_retry = config[DOMAIN][CONF_DISCOVERY_RETRY]
+    gateways = []
+    interface = 'any'
+    discovery_retry = 3
+    if DOMAIN in config:
+        gateways = config[DOMAIN][CONF_GATEWAYS]
+        interface = config[DOMAIN][CONF_INTERFACE]
+        discovery_retry = config[DOMAIN][CONF_DISCOVERY_RETRY]
+
+    def xiaomi_gw_discovered(service, discovery_info):
```

Comment:
Did you see the lint warning (C: 68, 4: Missing function docstring (missing-docstring))? Please just add the docstring.

Reply:
fixed